posts/2024-06-05-mixing-python-envs.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title: Building custom python environments on top of the hh5 conda analysis environments
+author: Dale Roberts
+categories: xarray
+---
+<script src="https://cdn.jsdelivr.net/gh/atteggiani/animated-terminal/animated-terminal.min.js" defer></script>
+
+# Building custom python environments on top of `conda/analysis3`
+
+## Introduction
+
+The conda analysis environments in the `hh5` project are a community-led approach to create a central 
+python environment specifically for the analysis of climate and weather data. As such, these environments
+contain a wide range of packages curated by the CLEX CMS team that can fulfil the requirements of most 
+analysis workflows. 
+
+CMS uses the following guidelines to determine whether a package can go into the `analysis3-unstable`
+environment:
+ * Is the package available from `conda-forge` or some other reputable conda channel?
+ * Is the package in line with the scope of the environment (i.e. climate data analysis)?
+ * Is the package actively maintained?
+ * Is the package and its dependencies compatible with the existing environment?
+
+If the answer to all of these questions is yes, then CMS will install the package into the environment on request. If, however, these conditions aren't met, that doesn't mean that you need to create a new conda environment
+from scratch to use a new package. 
+
+## `pip install --user`
+
+Lets use the [xWMB](https://github.com/hdrake/xwmb) package as an example.
+There is no conda distribution for this package, it can only be installed by `pip` from a git repository so it does not meet our criteria above. This
+package also has an extensive list of dependencies, most of which are available in the `conda/analysis3` environment, so using the `analysis3` 
+environment as a base will save quite a lot of disk space.
+ 
+You may be tempted to load the `conda/analysis3` environment and `pip install --user` the xWMB package.
```

Comment:
I wouldn't say this, for some people this might be the less messy option, and also sufficient as long as they don't need a fully reproducible option. So I would remove "You may be tempted.." and just discuss after then this is suitable and what are its limitations. For example for me, it's often fine because I sometimes just need to quickly install something to test user code and I don't care about what will happen in the future.